src/Compilers/CSharp/Portable/Symbols/NamespaceSymbol.cs

```diff
@@ -355,18 +355,23 @@ internal virtual void GetExtensionMethods(ArrayBuilder<MethodSymbol> methods, st
             }
         }
 
-        internal void GetExtensionContainers(ArrayBuilder<NamedTypeSymbol> extensions)
+        internal virtual void GetExtensionContainers(ArrayBuilder<NamedTypeSymbol> extensions)
         {
             foreach (var type in this.GetTypeMembersUnordered())
             {
-                if (!type.IsReferenceType || !type.IsStatic || type.IsGenericType || !type.MightContainExtensionMethods) continue;
+                AddExtensionContainersForType(type, extensions);
+            }
+        }
+
+        protected void AddExtensionContainersForType(NamedTypeSymbol type, ArrayBuilder<NamedTypeSymbol> extensions)
+        {
+            if (!type.IsReferenceType || !type.IsStatic || type.IsGenericType || !type.MightContainExtensionMethods) return;
```

Comment:
>IsReferenceType [](http://example.com/codeflow?start=22&length=15)

I think we can replace this with ```IsClassType``` to optimize this code even more.

Reply:
That's not an obvious equivalence (for me, at least). I'd prefer not to make that change, but will if you would like me to.

Reply:
Consider at least adding a comment capturing the suggestion and starting with ```// Tracked by [https://github.com/dotnet/roslyn/issues/76130](https://github.com/dotnet/roslyn/issues/76130) :```

Reply:
Done in commit 5